frontend/src/metabase-lib/lib/queries/StructuredQuery.ts

```diff
@@ -330,28 +330,31 @@ class StructuredQueryInner extends AtomicQuery {
     const sourceQuery = this.sourceQuery();
 
     if (sourceQuery) {
+      const fields = sourceQuery.columns().map(column => {
+        const id = [
+          "field",
+          column.name,
+          {
+            "base-type": column.base_type,
+          },
+        ];
+
+        return new Field({
+          ...column,
+          // TODO FIXME -- Do NOT use field-literal unless you're referring to a native query
+          id,
+          source: "fields",
+          // HACK: need to thread the query through to this fake Field
+          query: this,
+        });
+      });
+
       return new Table({
         id: this.sourceTableId(),
         name: "",
         display_name: "",
         db: sourceQuery.database(),
-        fields: sourceQuery.columns().map(
-          column =>
-            new Field({
-              ...column,
-              // TODO FIXME -- Do NOT use field-literal unless you're referring to a native query
-              id: [
-                "field",
-                column.name,
-                {
-                  "base-type": column.base_type,
-                },
-              ],
-              source: "fields",
-              // HACK: need to thread the query through to this fake Field
-              query: this,
-            }),
-        ),
+        fields,
```

Comment:
No changes here besides moving the evaluation of `fields` to a variable. I find myself stepping through this logic in the browser debugger not infrequently so having it be more debuggable is nice.

Reply:
Definitely way better.